test/database/query_builder.spec.ts

```diff
@@ -5439,6 +5439,55 @@ test.group('Query Builder | orderByRaw', (group) => {
   })
 })
 
+test.group('Query Builder | orderByRandom', (group) => {
+  group.setup(async () => {
+    await setup()
+  })
+
+  group.teardown(async () => {
+    await cleanup()
+  })
+
+  group.each.teardown(async () => {
+    await resetTables()
+  })
+
+  test('define order by random value', async ({ assert }) => {
+    const connection = new Connection('primary', getConfig(), logger)
+    connection.connect()
+
+    const db = getQueryBuilder(getQueryClient(connection))
+    await getInsertBuilder(getQueryClient(connection))
+      .table('users')
+      .multiInsert([
+        {
+          username: 'virk',
+          email: 'virk@adonisjs.com',
+        },
+        {
+          username: 'romain',
+          email: 'romain@adonisjs.com',
+        },
+        {
+          username: 'nikk',
+          email: 'nikk@adonisjs.com',
+        },
+      ])
+
+    const userResults: number[][] = []
+
+    for (let i = 0; i < 10; i++) {
+      const result = await db.from('users').orderByRandom()
+
+      userResults.push(result.map((user) => user.id))
+    }
```

Comment:
It looks like it was reusing the same query builder, so it would carry over the previous order by. 

I found that if you use a new query builder within the for loop, it works as expected: 

```suggestion
    const client = getQueryClient(connection)
    await getInsertBuilder(getQueryClient(connection))
      .table('users')
      .multiInsert([
        {
          username: 'virk',
          email: 'virk@adonisjs.com',
        },
        {
          username: 'romain',
          email: 'romain@adonisjs.com',
        },
        {
          username: 'nikk',
          email: 'nikk@adonisjs.com',
        },
      ])

    const userResults: number[][] = []

    for (let i = 0; i < 10; i++) {
      const result = await getQueryBuilder(client).from('users').orderByRandom()

      userResults.push(result.map((user) => user.id))
    }
```

Or clear the order by clauses like so:

```ts
const result = await db.from('users').clearOrder().orderByRandom()
```